File: 217. Contains Duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        map<int, bool> hash;

        for(int i=0;i<nums.size();i++){
            if(hash[nums[i]] == true)
                return true;
            hash[nums[i]] = true;
        }
        return false;
    }
};

File: Data Structure/Array & Strings/217. Contains Duplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        dic = set()

        for i, num in enumerate(nums):
            if num in dic:
                return True
            else:
                dic.add(num)
        return False
